code/datums/quirks/negative.dm

```diff
@@ -422,28 +422,68 @@
 /datum/quirk/social_anxiety/add()
 	RegisterSignal(quirk_holder, COMSIG_MOB_EYECONTACT, .proc/eye_contact)
 	RegisterSignal(quirk_holder, COMSIG_MOB_EXAMINATE, .proc/looks_at_floor)
+	RegisterSignal(quirk_holder, COMSIG_MOB_SAY, .proc/handle_speech)
 
 /datum/quirk/social_anxiety/remove()
 	UnregisterSignal(quirk_holder, list(COMSIG_MOB_EYECONTACT, COMSIG_MOB_EXAMINATE))
 
-/datum/quirk/social_anxiety/on_process(delta_time)
+/datum/quirk/social_anxiety/proc/handle_speech(datum/source, list/speech_args)
 	if(HAS_TRAIT(quirk_holder, TRAIT_FEARLESS))
 		return
+	var/datum/component/mood/mood = quirk_holder.GetComponent(/datum/component/mood)
```

Comment:
Mood component is not guaranteed (mood is in itself a config option). You need to handle the case where it does not exist.

Also, CC @ninjanomnom on GetComponent usage.

Reply:
Thinking about this.

If mood component isn't guaranteed due to config I'm more or less looking at it being limited to a *modifier* to `people_nearby` and I need to balance for `people_nearby` with a possibility of mood being null.

Not sure that will really add anything? I'm thinking like a 1.5 to 0.5 possible modifier from a combination of mood and sanity.

Reply:
Worst case, could just set Social Anxiety to be `mood_quirk = TRUE`, which means it can only be selected if mood is enabled in the config.

Reply:
GetComponent is justified, stop acting like it's the end of the world, that leads to stupid signal abuse

Reply:
Another option could be to use hunger as a modifier when mood is not available.

Reply:
I'm pretty openly "GetComponent is alright", though there's undeniably plenty of cases where it is not. That's why I CC'd ninjanomnom for an audit--getting the mood component is how most other code like this works.